src/integrationTest/kotlin/com/pubnub/api/integration/pam/GrantTokenIntegrationTest.kt

```diff
@@ -1,17 +1,71 @@
 package com.pubnub.api.integration.pam
 
 import com.pubnub.api.PubNub
+import com.pubnub.api.SpaceId
+import com.pubnub.api.UserId
 import com.pubnub.api.enums.PNLogVerbosity
 import com.pubnub.api.integration.BaseIntegrationTest
+import com.pubnub.api.models.consumer.access_manager.sum.SpaceIdGrant
+import com.pubnub.api.models.consumer.access_manager.sum.UserIdGrant
 import com.pubnub.api.models.consumer.access_manager.v3.ChannelGrant
 import com.pubnub.api.models.consumer.access_manager.v3.ChannelGroupGrant
 import com.pubnub.api.models.consumer.access_manager.v3.PNToken.PNResourcePermissions
-import org.junit.Assert
+import org.junit.Assert.assertEquals
 import org.junit.Test
 
 class GrantTokenIntegrationTest : BaseIntegrationTest() {
     private val pubNubUnderTest: PubNub = server
 
+    @Test
+    fun happyPath_SUM() {
+        // given
+        pubNubUnderTest.configuration.logVerbosity = PNLogVerbosity.BODY
+        val expectedTTL = 1337
+        val expectedAuthorizedUserId = UserId("authorizedUser01")
+        val expectedSpaceIdValue = "mySpace01"
+        val expectedSpaceIdPattern = "mySpace.*"
+        val expectedUserIdValue = "myUser01"
+        val expectedUserIdPattern = "myUser.*"
+
+        // when
+        val grantTokenEndpoint = pubNubUnderTest.grantToken(
+            ttl = expectedTTL,
+            authorizedUserId = expectedAuthorizedUserId,
+            spaces = listOf(
+                SpaceIdGrant.name(spaceId = SpaceId(expectedSpaceIdValue), read = true, delete = true),
+                SpaceIdGrant.pattern(pattern = expectedSpaceIdPattern, write = true, manage = true)
+            ),
+            userIds = listOf(
+                UserIdGrant.id(userId = UserId(expectedUserIdValue), delete = true),
+                UserIdGrant.pattern(pattern = expectedUserIdPattern, update = true)
+            )
+        )
+
+        val token = grantTokenEndpoint.sync()!!.token
+
+        // then
+        val (_, _, ttl, _, resources, patterns) = pubNubUnderTest.parseToken(token)
```

Comment:
oh... niceeee! It seems you're getting used to kotlin

Reply:
:)